libcloud/backup/base.py

```diff
@@ -0,0 +1,488 @@
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from libcloud.common.base import ConnectionUserAndKey, BaseDriver
+from libcloud.backup.types import BackupTargetType
+
+__all__ = [
+    'BackupTarget',
+    'BackupDriver',
+    'BackupTargetJob',
+    'BackupTargetRecoveryPoint'
+]
+
+
+class BackupTarget(object):
+    """
+    A backup target
+    """
+
+    def __init__(self, id, name, address, type, driver, extra=None):
+        """
+        :param id: Target id
+        :type id: ``str``
+
+        :param name: Name of the target
+        :type name: ``str``
+
+        :param address: Hostname, FQDN, IP, file path etc.
+        :type address: ``str``
+
+        :param type: Backup target type (Physical, Virtual, ...).
+        :type type: :class:`BackupTargetType`
+
+        :param driver: BackupDriver instance.
+        :type driver: :class:`BackupDriver`
+
+        :param extra: (optional) Extra attributes (driver specific).
+        :type extra: ``dict``
+        """
+        self.id = str(id) if id else None
+        self.name = name
+        self.address = address
+        self.type = type
+        self.driver = driver
+        self.extra = extra or {}
+
+    def update(self, name=None, address=None, extra=None):
+        return self.driver.update_target(target=self,
+                                         name=name,
+                                         address=address,
+                                         extra=extra)
+
+    def delete(self):
+        return self.driver.delete_target(target=self)
+
+    def _get_numeric_id(self):
+        target_id = self.id
+
+        if target_id.isdigit():
+            target_id = int(target_id)
+
+        return target_id
+
+    def __repr__(self):
+        return ('<Target: id=%s, name=%s, address=%s'
+                'type=%s, provider=%s ...>' %
+                (self.id, self.name, self.address,
+                 self.type, self.driver.name))
+
+
+class BackupTargetJob(object):
+    """
+    A backup target job
+    """
+
+    def __init__(self, id, status, progress, target, driver, extra=None):
+        """
+        :param id: Job id
+        :type id: ``str``
+
+        :param status: Status of the job
+        :type status: :class:`BackupTargetJobStatusType`
+
+        :param progress: Progress of the job, as a percentage
+        :type progress: ``int``
+
+        :param target: BackupTarget instance.
+        :type target: :class:`BackupTarget`
```

Comment:
Minor thing - since you are referring to a class in the same module you can use relative notation, so:

```python
:class:`.BackupTarget`
```

This way sphinx will know it refers to the class in the same module and will automatically link to it in the API docs.